storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1469,974 @@ public static int toPositive(int number) {
+    (io/copy (.getInputStream jarpath file) out)))))
+
+     */
+    public static void extractDirFromJar(String jarpath, String dir, String destdir) {
+        JarFile jarFile = null;
+        FileOutputStream out = null;
+        InputStream in = null;
+        try {
+            jarFile = new JarFile(jarpath);
+            Enumeration<JarEntry> jarEnums = jarFile.entries();
+            while (jarEnums.hasMoreElements()) {
+                JarEntry entry = jarEnums.nextElement();
+                if (!entry.isDirectory() && entry.getName().startsWith(dir)) {
+                    File aFile = new File(destdir, entry.getName());
+                    aFile.getParentFile().mkdirs();
+                    out = new FileOutputStream(aFile);
+                    in = jarFile.getInputStream(entry);
+                    IOUtils.copy(in, out);
+                    out.close();
+                    in.close();
+                }
+            }
+        } catch (IOException e) {
+            LOG.info("Could not extract {} from {}", dir, jarpath);
+        } finally {
+            if (jarFile != null) {
+                try {
+                    jarFile.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                            "Something really strange happened when trying to close the jar file" + jarpath);
+                }
+            }
+            if (out != null) {
+                try {
+                    out.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                            "Something really strange happened when trying to close the output for jar file" + jarpath);
+                }
+            }
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                            "Something really strange happened when trying to close the input for jar file" + jarpath);
+                }
+            }
+        }
+
+    }
+
+    public static int sendSignalToProcess(long pid, int signum) {
+        int retval = 0;
+        try {
+            String killString = null;
+            if (isOnWindows()) {
+                if (signum == SIGKILL) {
+                    killString = "taskkill /f /pid ";
+                } else {
+                    killString = "taskkill /pid ";
+                }
+            } else {
+                killString = "kill -" + signum + " ";
+            }
+            killString = killString + pid;
+            retval = execCommand(killString);
+        } catch (ExecuteException e) {
+            LOG.info("Error when trying to kill " + pid + ". Process is probably already dead.");
+        } catch (IOException e) {
+            LOG.info("IOException Error when trying to kill " + pid + ".");
+        } finally {
+            return retval;
```

Comment:
The original code would throw an IOException, but catch and ExecuteException.  We should follow the same pattern for now.

Also lets just remove retval.  The ExecuteException should already be handling checking for errors, so returning things is just going to cause confusion.